source/common/local_reply/local_reply.cc

```diff
@@ -26,7 +26,7 @@ class BodyFormatter {
             config.format_case() ==
                     envoy::config::core::v3::SubstitutionFormatString::FormatCase::kJsonFormat
                 ? Http::Headers::get().ContentTypeValues.Json
-                : Http::Headers::get().ContentTypeValues.Text) {}
+                : Formatter::SubstitutionFormatStringUtils::getContentType(config)) {}
```

Comment:
So, given: https://github.com/envoyproxy/envoy/pull/13019/files#r488273786, here we can have: `config.content_type().empty() ? Http::Headers::get().ContentTypeValues.Text : config.content_type()`.